pkg/didcomm/common/service/destination_default.go

```diff
@@ -70,7 +74,7 @@ func CreateDestination(didDoc *diddoc.Doc) (*Destination, error) {
 	} else { // didcomm v1 service
 		didCommService, ok = diddoc.LookupService(didDoc, didCommServiceType)
 		if !ok {
-			return nil, fmt.Errorf("create destination: missing DID doc service")
+			return createLegacyDestination(didDoc)
```

Comment:
Maybe as a part of incremental refactoring we could split the whole `CreateDestionation` method onto `createV2`, `createV1` and `createLegacy`? To avoid long nested if's